CHANGELOG.md

```diff
@@ -2,6 +2,506 @@
+
+```json
+{
+  "errors": [{ "message": "Some execution error" }]
+}
+```
+
+This fix ensures full compliance with the GraphQL specification by preserving the complete structure of error responses from coprocessors.
+
+Contributed by [@IvanGoncharov](https://github.com/IvanGoncharov) in [#7141](https://github.com/apollographql/router/pull/7141)
+
+### Improve Error Message for Invalid JWT Header Values ([PR #7121](https://github.com/apollographql/router/pull/7121))
+
+Enhanced parsing error messages for JWT Authorization header values now provide developers with clear, actionable feedback while ensuring that no sensitive data is exposed.
+
+Examples of the updated error messages:
+```diff
+-         Header Value: '<invalid value>' is not correctly formatted. prefix should be 'Bearer'
++         Value of 'authorization' JWT header should be prefixed with 'Bearer'
+```
+
+```diff
+-         Header Value: 'Bearer' is not correctly formatted. Missing JWT
++         Value of 'authorization' JWT header has only 'Bearer' prefix but no JWT token
+```
+
+By [@IvanGoncharov](https://github.com/IvanGoncharov) in https://github.com/apollographql/router/pull/7121
+
+### Helm: Correct default telemetry `resource` property in `ConfigMap` (copy #6105) ([Issue #6104](https://github.com/apollographql/router/issues/6104))
+
+The Helm chart was using an outdated value when emitting the `telemetry.exporters.metrics.common.resource.service.name` values.  This has been updated to use the correct (singular) version of `resource` (rather than the incorrect `resources` which was used earlier in 1.x's life-cycle).
+
+By [@vatsalpatel](https://github.com/vatsalpatel) in https://github.com/apollographql/router/pull/6105
+
+### Update Dockerfile exec script to use `#!/bin/bash` instead of `#!/usr/bin/env bash` ([Issue #3517](https://github.com/apollographql/router/issues/3517))
+
+For users of Google Cloud Platform (GCP) Cloud Run platform, using the router's default Docker image was not possible due to an error that would occur during startup:
+
+```sh
+"/usr/bin/env: 'bash ': No such file or directory"
+```
+
+To avoid this issue, we've changed the script to use `#!/bin/bash` instead of `#!/usr/bin/env bash`, as we use a fixed Linux distribution in Docker which has the Bash binary located there.
+
+By [@lleadbet](https://github.com/lleadbet) in https://github.com/apollographql/router/pull/7198
+
+### Remove "setting resource attributes is not allowed" warning ([PR #7272](https://github.com/apollographql/router/pull/7272))
+
+If Uplink is enabled, Router 2.1.x emits this warning at startup event though no user configuration or other choice is responsible for it:
+
+```
+WARN  setting resource attributes is not allowed for Apollo telemetry
+```
+
+This removes the warning entirely as it’s not particularly helpful.
+
+Reproduction:
+
+```
+APOLLO_KEY=secret APOLLO_GRAPH_REF=starstuff@current cargo run
+```
+
```

Comment:
I don't think this reproduction matters to anyone reading the changelog 😄 

````suggestion
Reproduction:

```
APOLLO_KEY=secret APOLLO_GRAPH_REF=starstuff@current cargo run
```

````